spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesDiscoveryProperties.java

```diff
@@ -188,15 +187,10 @@ public record Metadata(@DefaultValue("true") boolean addLabels, String labelsPre
 			@DefaultValue("true") boolean addAnnotations, String annotationsPrefix,
 			@DefaultValue("true") boolean addPorts, @DefaultValue("port.") String portsPrefix) {
 
-		@ConstructorBinding
-		public Metadata {
-
-		}
-
-		// needed in order to get the defaults for some fields
-		public Metadata() {
-			this(true, null, true, null, true, "port.");
-		}
+		/**
+		 * Default instance.
+		 */
+		public static final Metadata DEFAULT = new Metadata(true, null, true, null, true, "port.");
```

Comment:
having a default instance, is easier to manage and understand and I did this in one of the PRs before.